hypothesis-python/src/hypothesis/internal/conjecture/data.py

```diff
@@ -715,10 +717,15 @@ def __init__(
         self._observability_predicates: defaultdict[str, PredicateCounts] = defaultdict(
             PredicateCounts
         )
+
         self._sampled_from_all_strategies_elements_message: Optional[
             tuple[str, object]
         ] = None
         self._shared_strategy_draws: dict[Hashable, tuple[int, Any]] = {}
+        self._shared_data_strategy: Optional[DataObject] = None
+        self._stateful_repr_parts: Optional[list[Any]] = None
+        self.states_for_ids: Optional[dict[int, RandomState]] = None
+        self.seeds_to_states: Optional[dict[Any, RandomState]] = None
         self.hypothesis_runner = not_set
```

Comment:
Adding these attributes is mainly in anticipation of mypyc, where it will be required.